net/netxng/src/TNetXNGFile.cxx

```diff
@@ -131,14 +131,11 @@ TNetXNGFile::TNetXNGFile(const char *url,
                          Bool_t      parallelopen) :
 	TNetXNGFile(url,0,mode,title,compress,netopt,parallelopen){}
 
-TNetXNGFile::TNetXNGFile(const char *url,
-		         const char *lurl,
-                         Option_t   *mode,
-                         const char *title,
-                         Int_t       compress,
-                         Int_t       /*netopt*/,
-                         Bool_t      parallelopen) :
-   TFile((lurl ? lurl : url), "NET", title, compress)
+TNetXNGFile::TNetXNGFile(const char *url, const char *lurl, Option_t *mode, const char *title, Int_t compress,
+                         Int_t /*netopt*/, Bool_t parallelopen)
+   : TFile((lurl ? lurl : url),
+           strcmp(mode, "READ_WITHOUT_GLOBALREGISTRATION") == 0 ? "NET_WITHOUT_GLOBALREGISTRATION" : "NET", title,
```

Comment:
```suggestion
           strstr(mode, "_WITHOUT_GLOBALREGISTRATION") != nullptr ? "NET_WITHOUT_GLOBALREGISTRATION" : "NET", title,
```
would allow to seamlessly support the other opening type.

Reply:
what other opening type? `NET_WITHOUT_GLOBALREGISTRATION` is not available to client code, it's just for internal use (and it's not documented anywhere). the documented option is only `READ_WITHOUT_GLOBALREGISTRATION`. I think it's simpler this way? (also considering the behavior before this patch)

Reply:
With the change on line 364 and related, we now support (even if not documented yet) all combination of opening type and `_WITHOUT_GLOBALREGISTRATION`, including `RECREATE_WITHOUT_GLOBALREGISTRATION`, `UPDATE_WITHOUT_GLOBALREGISTRATION` etc.  And I think this is a good thing (and should be documented of course :) ).
And without the proposed, change we keep at supporting 'only' `READ_WITHOUT_GLOBALREGISTRATION` for 'only' the`NET/WEB` case.

Reply:
adding support for `RECREATE_WITHOUT_GLOBALREGISTRATION` etc., adding documentation for them and the corresponding tests is beyond the scope of this PR. The intent is just to fix https://github.com/root-project/root/issues/10742 .

Reply:
But to be consistent, you would have to add code to explicitly reject the new supported mode; where as extending is 'simply' using the `strstr` and extending the doc which is currently bare:
```
204:/// READ_WITHOUT_GLOBALREGISTRATION   | Used by TTreeProcessorMT, not a user callable option.
```
and could simply be as bare:
```
204:/// [READ|NEW|UPDATE|RECREATE]_WITHOUT_GLOBALREGISTRATION   | Used by TTreeProcessorMT, not a user callable option.
```

Reply:
eg. TTreeProcessorMT could use `RECREATE_WITHOUT_GLOBALREGISTRATION` for `Snapshot` output file.

Reply:
> to be consistent, you would have to add code to explicitly reject the new supported mode

That's not the behavior anywhere in TFile:

```cpp
$ root -l
root [0] TFile::Open("f.root", "alskjdalksjd")
(TFile *) 0x559fa18d0200
root [1] TFile::Open("root://eospublic.cern.ch//eos/root-eos/cms_opendata_2012_nanoaod_skimmed/ZZTo4mu.root", "alsjalsjdalsdjka")
(TFile *) 0x555fc11ba680
```

I just picked up https://github.com/root-project/root/issues/10742 although it was not assigned to me because that specific issue is problematic for RDF. Adding new features to TFile that were not requested or discussed before can be done in a different PR, and it does not seem reasonable to me to request them from me in the context of a bug fix.

With that said, I applied your suggestion in the original comment so it will be easier to extend this in the future.